test/parallel/test-fs-write-stream-autoclose-option.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const path = require('path');
+const fs = require('fs');
+
+var file = path.join(common.tmpDir, 'write-autoclose-opt1.txt');
+common.refreshTmpDir();
+var stream = fs.createWriteStream(file, {flags: 'w+', autoClose: false});
+stream.write('Test1');
+stream.end();
+stream.on('finish', common.mustCall(function() {
+  process.nextTick(common.mustCall(function() {
+    assert.strictEqual(stream.closed, undefined);
+    assert.strictEqual(!stream.fd, false);
```

Comment:
if you're specifically testing for not null, perhaps `assert(stream.fd !== null)` would be better